Extensions/DialogueTree/JsExtension.js

```diff
@@ -0,0 +1,516 @@
+ * to this extension file or to any other *.js file that you reference inside.
+ *
+ * The file must be named "JsExtension.js", otherwise GDevelop won't load it.
+ * ⚠️ If you make a change and the extension is not loaded, open the developer console
+ * and search for any errors.
+ *
+ * More information on https://github.com/4ian/GDevelop/blob/master/newIDE/README-extensions.md
+ */
+module.exports = {
+	createExtension: function(_, gd) {
+		const extension = new gd.PlatformExtension();
+		extension
+			.setExtensionInformation(
+				'DialogueTree',
+				_('Dialoguetree'),
+				_('Initiate tree dialogues from Yarn via Bondagejs.'),
+				'Todor Imreorov',
+				'Open source (MIT License)'
+			)
+			.setExtensionHelpPath('/all-features/dialoguetree');
+
+		extension
+			.addAction(
+				'LoadDialogueFromSceneVariable',
+				_('Load Dialogue Tree from a Scene Variable'),
+				_('Load a dialogue data object - Yarn json format, stored in a scene variable'),
+				_('Load Dialogue data from Scene Variable: _PARAM1_ and begin from _PARAM2_ Node'),
+				_('Dialogue Tree'),
+				'JsPlatform/Extensions/yarn24.png',
+				'JsPlatform/Extensions/yarn32.png'
+			)
+			.addCodeOnlyParameter('currentScene', '')
+			.addParameter('scenevar', _('Scene variable that holds the Yarn Json data'))
+			.addParameter('string', _('Dialogue node'))
+			.setDefaultValue('Start')
+			.getCodeExtraInformation()
+			.setIncludeFile('Extensions/DialogueTree/dialoguetools.js')
+			.addIncludeFile('Extensions/DialogueTree/bondage.min.js')
+			.setFunctionName('gdjs.dialogueTree.loadFromSceneVar');
+
+		extension
+			.addAction(
+				'LoadDialogueFromJsonFile',
+				_('Load Dialogue Tree from a Json File'),
+				_('Load a dialogue data object - Yarn json format, stored in a Json file'),
+				_('Load Dialogue data from Json File: _PARAM1_ and begin from _PARAM2_ Node'),
+				_('Dialogue Tree'),
+				'JsPlatform/Extensions/yarn24.png',
+				'JsPlatform/Extensions/yarn32.png'
+			)
+			.addCodeOnlyParameter('currentScene', '')
+			.addParameter('jsonResource', _('Json file that holds the Yarn Json data'))
+			.addParameter('string', _('Dialogue node'))
+			.getCodeExtraInformation()
+			.setIncludeFile('Extensions/DialogueTree/dialoguetools.js')
+			.addIncludeFile('Extensions/DialogueTree/bondage.min.js')
+			.setFunctionName('gdjs.dialogueTree.loadFromJsonFile');
+
+		extension
+			.addAction(
+				'StarDialogueFromNode',
+				_('Start Dialogue from Node'),
+				_('Start Dialogue from Node in Yarn. You can use this to store multiple Npcs in one Yarn file'),
+				_('Start Dialogue from Node: _PARAM0_'),
+				_('Dialogue Tree'),
+				'JsPlatform/Extensions/yarn24.png',
+				'JsPlatform/Extensions/yarn32.png'
+			)
+			.addParameter('string', _('Dialogue node'))
+			.getCodeExtraInformation()
+			.setFunctionName('gdjs.dialogueTree.startFrom');
+
+		extension
+			.addAction(
+				'AdvanceToNextLine',
+				_('Advance to the Next Dialogue Line'),
```

Comment:
I think "Advance" emphasizes the transition from one state to the next one in line a bit better than a mere "go to". But it might just be the programmer's way of thinking. When reading "GOTO" I immediately think of spaghetti code. ^^

Reply:
I kind of used advance exactly because of what @Wend1go describes. But I can rename it, no probs :)